### DISCUSSION TITLE
How to config to let L2TP/IPsec work correctly ?

### DISCUSSION THREAD

**Discussion by carasha** [accel-ppp/accel-ppp#168] [L2TP] [answered]
My OS : CentOS 7

I can bulid L2TP/IPsec ( xl2ptd + strongswan , using Pre-Shared-key ) VPN Server , it's no problem at all .

I am new in accel-ppp , i can build PPTP Server with accel-ppp , and it works fine .

Now , i want to use accel-ppp to build L2TP/IPsec VPN Server , but i have no idea how to config L2TP + IPSec in accel-ppp .

Can any one help me ? or give me a hint ?

What i think is : 

xl2tpd + strongswan ( work fine )

accel-ppp + strongswan 

So , i just keep strongswan no change ( don't make any change on strongswan) , and use accel-ppp's L2TP to replace xl2tpd , is that correct ?

**Comment by DmitriyEshenko** [comment]
Hi @carasha I think it is preaty similar like xl2tpd, try to check how VyOS it is do https://github.com/vyos/vyos-1x/blob/current/data/templates/accel-ppp/l2tp.config.j2#L41-L91

**Reply by carasha** [reply, marked as answer]
Thank you for your comment .

What i think is correct  ( keep strongswan no change and use accel-ppp's l2tp to replace xl2tpd ) .

If your xl2tpd + strongswan works good , then you want to replace xl2tpd by accel-ppp's l2tp :
Just keep strongswan's config no change , and use accel-ppp's l2tp to replace xl2tpd , it can work .
